src/py42/clients/settings/device_settings.py

```diff
@@ -0,0 +1,415 @@
+            self, "excluded_files", excludes, self._changes
+        )
+        self._filename_exclusions = TrackedFileSelectionList(
+            self, "filename_exclusions", regex_excludes, self._changes
+        )
+        self._orig_destinations = self.destinations
+
+    @property
+    def included_files(self):
+        """Returns the list of files/folders included in the backup selection. Items can
+        be added/removed from this list via normal list methods, or assigning a new list
+        of files to this attribute to replace the existing one.
+        """
+        return self._included_files
+
+    @included_files.setter
+    def included_files(self, value):
+        if isinstance(value, (list, tuple)):
+            self._included_files.clear()
+            self._included_files.extend(value)
+        else:
+            raise AttributeError("included files must be a list/tuple.")
+
+    @property
+    def excluded_files(self):
+        """Returns the list of files/folders excluded from the backup selection. Items can
+        be added/removed from this list via normal list methods, or assigning a new list
+        of files to this attribute to replace the existing one.
+        """
+        return self._excluded_files
+
+    @excluded_files.setter
+    def excluded_files(self, value):
+        if isinstance(value, (list, tuple)):
+            self._excluded_files.clear()
+            self._excluded_files.extend(value)
+        else:
+            raise AttributeError("excluded files must be a list/tuple.")
+
+    @property
+    def filename_exclusions(self):
+        """Returns the list of regex patterns used to exclude file paths from the backup
+        selection. Items can be added/removed from this list via normal list methods,
+        or assigning a new list of patterns to this attribute to replace the existing
+        one.
+        """
+        return self._filename_exclusions
+
+    @filename_exclusions.setter
+    def filename_exclusions(self, value):
+        if isinstance(value, (list, tuple)):
+            self._filename_exclusions.clear()
+            self._filename_exclusions.extend(value)
+        else:
+            raise AttributeError("filename exclusions must be a list/tuple.")
+
+    @property
+    def destinations(self):
+        """Returns a dict of the destinations used for backup for the backup set. Dict
+        keys are the destination guids, values are the destination names.
+        """
+        destination_dict = {}
+        for d in self.data["destinations"]:
+            guid = d["@id"]
+            dest_name = self._manager.available_destinations[guid]
+            if "@locked" in d:
+                dest_name = dest_name + " <LOCKED>"
+            destination_dict[guid] = dest_name
+        return destination_dict
+
+    def add_destination(self, destination_guid):
+        """Adds a destination to be used by this backup set. Raises an exception if
```

Comment:
would be better to specify the exception class here

Reply:
done